homework/index.js

```diff
@@ -30,13 +30,106 @@
+      });
+      createAndAppend('td', descriptionRow, { text: repo.description });
+    }
+
+    const forksRow = createAndAppend('tr', tBody, { class: 'row' });
+    createAndAppend('td', forksRow, { text: 'Forks :', class: 'label' });
+    createAndAppend('td', forksRow, { text: repo.forks });
+    const dateRow = createAndAppend('tr', tBody, { class: 'row' });
+    createAndAppend('td', dateRow, {
+      text: 'Updated :',
+      class: 'label',
+    });
+    createAndAppend('td', dateRow, {
+      text: new Date(repo.updated_at).toLocaleString(),
+    });
+  }
+
+  function reposContributor(url) {
+    const root = document.getElementById('root');
+    const rightDiv = document.querySelector('.rightDiv');
+    createAndAppend('p', rightDiv, {
+      class: 'contributor-header',
+      text: 'Contributions',
+    });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    fetchJSON(url, (err, contributors) => {
+      if (err) {
+        createAndAppend('p', root, { text: err.message, class: 'alert' });
+      } else {
+        contributors.forEach(contributor => {
+          const li = createAndAppend('li', ul, {
+            class: 'contributor-item',
+            tabindex: '0',
+            'aria-label': contributor.login,
+          });
+
+          createAndAppend('img', li, {
+            src: contributor.avatar_url,
+            height: 48,
+            class: 'contributor-avatar',
+          });
+          const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+          createAndAppend('div', contributorData, { text: contributor.login });
+          createAndAppend('div', contributorData, {
+            class: 'contributor-badge',
+            text: contributor.contributions,
+          });
+
+          li.addEventListener('click', () => {
+            window.open(contributor.html_url);
+          });
+        });
+      }
+    });
+  }
+
   function main(url) {
+    const root = document.getElementById('root');
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
+      const header = createAndAppend('header', root, { class: 'header' });
+      createAndAppend('p', header, { text: 'HYF Repositories' });
+      const selector = createAndAppend('select', header, { class: 'repo-selector' });
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        createAndAppend('p', root, { text: err.message, class: 'alert' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        data.sort((a, b) => a.name.toLowerCase().localeCompare(b.name.toLowerCase()));
+        for (let i = 0; i < data.length; i++) {
+          createAndAppend('option', selector, { text: data[i].name, value: i });
+        }
+
+        const container = createAndAppend('div', root, { id: 'container' });
+        createAndAppend('div', container, { class: 'leftDiv' });
```

Comment:
It is a best practice to use kebab-case for CSS names: `leftDiv` -> `left-div`.

Reply:
thank you..